akirachix/Ciphers-Frontend
# Added a boiler plate

What this PR does?

What is this?  
This PR updates the Totosteps dashboard, introducing new features such as User Analysis, Resource Usage, and Development tracking, along with UI enhancements to existing modules like Users and Assessment. These additions are aimed at improving admin workflow and providing insights into system performance and user behavior.

Why this?  
The Totosteps dashboard will require a tool for admins to manage user data, track system resources, and oversee development progress efficiently. These updates will be driven by the need to enhance performance, streamline data visualization, and improve admin decision-making. 

How are you doing this?
- User Analysis: A new feature to track detailed user behavior and interactions.
- Resource Usage: A module for monitoring system resources and usage trends.
- Development: A feature to track ongoing project progress and system updates.
- UI/UX Enhancements: Improvements to the Users and Assessment sections for better data representation and faster loading times.

List of updates:
- Added User Analysis section for user tracking.
- Added Resource Usage section to monitor system performance.
- Added Development tracking for project monitoring.
- Optimized Users and Assessment modules for better data handling.
- Bug fixes 


Type of Change




 How can it be tested?

1. Step-by-step Guide:
   - Log in as an admin and navigate to the User Analysis, Resource Usage, and Development sections from the sidebar.
   - Test real-time data rendering and loading times in the Users and Assessment sections.
   - Verify the correct display of user interaction data in the User Analysis module.
   - Monitor resource usage data in the Resource Usage section.


Where is it documented?

- JIRA Ticket
